### PR TITLE
Fix Long Title Text

### LINKED ISSUE #54: BUG: Titles, Tags, or Categories with long words cause horizontal scrolling small-width devices.
# Bug Report
## Describe the bug
I'm not sure if this is actually a bug or working as intended. I also apologize in advance if I use the wrong terminology for some things. Web dev is not my area of expertise.

When I create posts that contain Titles, Tags, or Categories that contain long words, the page gets rendered wider than the viewport(?) on devices with a relatively small width such as a phone. By long words, I mean something like `thisisalongwordwithnospaces`. Adding spaces or hyphens allows the words to wrap on those boundaries and the view renders normally again.

## To Reproduce

1. Create a new post that contains any or all of the following. The longer the word is, the more obvious the problem is.

* `title = "My Post WithASuperLongWordIsLong"`
* `categories = ["MySuperLongWordIsLong"]`
* `tags = ["mysuperlongwordislong"]`

2. Open the page on a phone or emulated iPhone 6/7/8 view in Chrome

## Expected behavior

For Categories and Tags, I'd probably prefer the words get cut off (clipped?) rather than cause the whole page to be rendered too wide and need horizontal scrolling. I have no idea how to make this happen though. I tried playing around with overflow-related CSS things in Chrome dev tools without much luck.

I'm not exactly sure what I'd prefer to happen for Titles. Is there a way to force the browser to wrap long words even if they don't have natural breaks like a hyphen?

## Environment

 - **Hugo Version:** 0.57.0
 - **Browser and Version:** Mobile Safari on iOS 12.4, Chrome 76.0.3809.100 emulating iPhone 6/7/8, 
 - **Platform:** Mobile
 - **Link to Project:**

### LINKED ISSUE #91: BUG: pagination button size on long titles
# Bug Report
## Describe the bug
The buttons to navigate to the next or previous article are designed for quite short titles. Longer titles get wrapped. In this case, the border is to small. 

now

In my opinion, you can fix it by allowing the box to grow:
Currently, the css class button contains "`height: 4.8125em;`" . By switching to "`min-height: 4.8125em;`" the box can grow. I think, for usability, that's good solution.
min-height

A second way would be to cut the text. This could be done by adding these attributes to button:
```
overflow: hidden;
white-space: nowrap;
text-overflow: ellipsis;
```

From a designers point of view, this is probably preferable. Even better would be to cut the title and keep the ">". 

overflow-hidden

## Environment

 - **Hugo Version:** v0.57.2
 - **Browser and Version:** Firefox 68, Chrome 76
 - **Platform:** Dektop

### PR DESCRIPTION
**Pull request by pacollins**
## Description

Truncates titles, replaces with ellipse.

## Motivation and Context

Fixes #91 👍 - Done

Fixes #54 👍 - Done

## Screenshots (if appropriate):

For #91:
large
small

For #54:

Tags and categories function the same as above. Header just wraps.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have read the [Contributing Document](https://github.com/pacollins/hugo-future-imperfect-slim/blob/master/.github/CONTRIBUTING.md).
- [X] My code follows the [code style](https://github.com/pacollins/hugo-future-imperfect-slim/blob/master/.github/CONTRIBUTING.md#Style-Guide) of this project.
- [ ] My change requires a change to the [documentation](https://github.com/pacollins/hugo-future-imperfect-slim/wiki).
- [ ] I have updated the documentation, including `theme.toml`, accordingly.

### CODE REVIEW

File: layouts/post/pagination.html [hunk 1]
```diff
@@ -1,8 +1,8 @@
 <div class="pagination">
   {{ if .PrevInSection }}
-    <a href="{{ .PrevInSection.RelPermalink }}" class="button big previous"><i class="fas fa-angle-left"></i> {{ .PrevInSection.Title }}</a>
+    <a href="{{ .PrevInSection.RelPermalink }}" class="button"><div class="previous"><div>{{ .PrevInSection.Title }}</div></div></a>
```
Comment (pacollins) [inline layouts/post/pagination.html]: Open to a more elegant solution than throwing another `<div>` in here.

File: assets/scss/main.scss [hunk 2]
```diff
@@ -1015,6 +1015,7 @@ textarea {
      border-bottom: solid 1px rgba(160, 160, 160, 0.3);
      margin: 1em 1em 0 1em;
      text-align: center;
+     word-break: break-all;
```
Comment (pacollins) [inline assets/scss/main.scss]: I really would like to try to hyphenate, but I couldn't seem to get it working.

Comment (th1l6f) [reply assets/scss/main.scss]: The introduction of `word-break: break-all;` causes grammatically incorrect word breaks in the post's title. The title is the first impression of possible reader. IMHO that's not a good place for mistakes.

I've set `word-break: initial;` in my `add-on.ccs`. At first glance, I don't see any cons.

Comment (pacollins) [reply assets/scss/main.scss]: See #99

**Pull request merged**